src/controller/reco_controller.py

```diff
@@ -222,56 +238,61 @@ def get_items(self) -> tuple[list, list[list], str]:
                 res.append(items[0])
             return list(selected_models), res, self.model_config
 
-    def get_items_by_strategy_and_model(
-        self, model_info: dict
-    ) -> tuple[list, list[list], str]:
-        """Gets all items from OSS by model info
-
-        Gets the start items by model info. The start item is the reference item a user searches.
-        Each start item creates a new row in the frontend.
+    def get_items_by_strategy_and_model(self, model_info: dict) -> tuple[list, list[list], str]:
+        """
+        Gets start item(s) based on model info and already set strategy. If model is configured as
+        recos_in_same_response, function will not only return start items but also all recommended
+        items as well. Otherwise function will try to get recommended results for start item.
 
-        :param model_info: Config of model in configuration yaml
-        :return:
+        :param model_info: selected model info dictionary
+        :return:found items in a list
         """
         all_items = []
         item_hits, start_items = self._get_start_items(model_info)
 
-        self.set_num_pages(item_hits)
-
-        for index, start_item in enumerate(start_items):
+        if model_info.get('recos_in_same_response', False):
             item_row = []
-            item_row.append(start_item)
-            try:
-                nn_items, nn_dists = self._get_reco_items(start_item, model_info)
-
-                # Find all filters
-                all_chosen_filters = self._get_current_filter_state("reco_filter")
-                if len(all_chosen_filters["remove_duplicate"]) > 0:
-                    chosen_param = []
-                    for item in all_chosen_filters["remove_duplicate"]:
-                        chosen_param.append(item[1])
-                    filtered_nn_items = self.postproc.filterDuplicates(
-                        start_item, nn_items, chosen_param
+            for index, start_item in enumerate(start_items):
+                item_row.append(start_item)
+            all_items.append(item_row)
+            return [model_info['display_name']], all_items, self.model_config
+        else:
+            self.set_num_pages(item_hits)
```

Comment:
wir haben hier echt viel logik in einer Funktion, wenn du noch Lust hast, schlage ich vor beide Strategien in eigene Funktionen auszulagern.